Bio/Align/AlignInfo.py

```diff
@@ -178,6 +179,144 @@ def gap_consensus(self, threshold=.7, ambiguous="X",
 
         return Seq(consensus, consensus_alpha)
 
+    def iupac_consensus(self, ambiguous="X", 
+                        consensus_alpha=None, require_multiple=0):
+        """Output a fast consensus sequence of the DNA alignment.
+
+        This doesn't do anything fancy at all. It will just go through the
+        sequence residue by residue(ie. A or G or T or C for DNA) 
+        in all sequences in the alignment. It will add IUPAC nucleotide 
+        ambiguity bases where appliciable. This does not account for number
+        of bases, if only 1 base differs in an alignment of 100, 
+        This will will call an ambiguity. A very conservative approach. 
+
+        Arguments:
+         - ambiguous - The ambiguous character to be added when the threshold is
+           not reached.
+         - consensus_alpha - The alphabet to return for the consensus sequence.
+           If this is None, then we will try to guess the alphabet.
+         - require_multiple - If set as 1, this will require that more than
+           1 sequence be part of an alignment to put it in the consensus (ie.
+           not just 1 sequence and gaps).
+
+        """
+        # Mike Lloyd, 11-OCT-2017: modified dumb_consensus to add iupac bases
+        
+        inv_ambig = {v: k for k, v in IUPACData.ambiguous_dna_values.items()}
+
+        consensus = ''
+
+        # find the length of the consensus we are creating
+        con_len = self.alignment.get_alignment_length()
+
+        # go through each seq item
+        for n in range(con_len):
+            # keep track of the counts of the different atoms we get
+            atom_dict = {}
+            num_atoms = 0
+
+            for record in self.alignment:
+                # make sure we haven't run past the end of any sequences
+                # if they are of different lengths
+                if n < len(record.seq):
+                    if record.seq[n] != '-' and record.seq[n] != '.':
+                        if record.seq[n] not in atom_dict:
+                            atom_dict[record.seq[n]] = 1
+                        else:
+                            atom_dict[record.seq[n]] += 1
+
+                        num_atoms = num_atoms + 1
+
+            max_atoms = []
+            max_size = 0
+
+            for atom in atom_dict:
+                max_atoms.append(atom)
+                #Generate a list of all bases at residue 
+                #no weighting/counting/threshold as in dumb_consensus
+
+            if require_multiple and num_atoms == 1:
+                consensus += ambiguous
+            elif (len(max_atoms) == 1):
```

Comment:
Too many brackets for preferred Python style, use:

```python
elif len(max_atoms) == 1:
```

Reply:
Corrected in 6b9a569